pkg/opendsme/contrib/DSMEPlatform.cpp

```diff
@@ -0,0 +1,738 @@
+/*
+ * Copyright (C) 2022 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  José I. Álamos <jose.alamos@haw-hamburg.de>
+ */
+
+#include "opendsme/opendsme.h"
+#include "opendsme/DSMEPlatform.h"
+#include "ztimer.h"
+#include "iolist.h"
+#include "event.h"
+#include "event/thread.h"
+#include "luid.h"
+#include "dsmeAdaptionLayer/scheduling/TPS.h"
+#include "dsmeAdaptionLayer/scheduling/StaticScheduling.h"
+#include "board.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+/* Use default openDSME value for TPS scheduler alpha */
+#define OPENDSME_TPS_ALPHA (0.1)
+
+/* Used for symbol counter calculation */
+#define OPENDSME_TIMER_MASK   (0xF)
+#define OPENDSME_TIMER_OFFSET (4U)
+
+namespace dsme {
+
+/* The one and only openDSME instance */
+DSMEPlatform *DSMEPlatform::instance = nullptr;
```

Comment:
Does that mean we can't have multiple interfaces currently?

Reply:
Yup. Although it should be possible from a code perspective, running two concurrent scheduled MAC layers would be tricky for most devices. But it's possible in theory.
This does not prevent having other interfaces (eth, submac, etc)